### PR TITLE
Add traits to JWT payload.

### LINKED ISSUE #13267: Provide more information in the JWT payload in application access.
**What would you like Teleport to do?**
Currently the JWT payload header included in the application access headers contains the username and their roles in teleport. It would be useful if it could also contain any traits assigned to that user. 

**What problem does this solve?**
It would allow for much more granular control of how to handle the user on the application side. For example, two users might have a trait called 'department' and be assigned a role called 'department-access' which allows access to any nodes with a 'department' node label that matches their trait. That works perfectly within teleport, but without the traits being included in the JWT payload there's no way to achieve that level of granularity from within an application.

**If a workaround exists, please include it.**
None that I was able to find.

### PR DESCRIPTION
**Pull request by mdwn**
The JWT payload now includes user traits.

Example of new payload:

```
{
  "aud": [
    "http://127.0.0.1:9000"
  ],
  "exp": 1665124276,
  "iat": 1665081320,
  "iss": "mw-test.teleport",
  "nbf": 1665081310,
  "roles": [
    "access",
    "editor"
  ],
  "sub": "teleport-admin",
  "traits": {
    "logins": [
      "root",
      "ubuntu",
      "ec2-user"
    ]
  },
  "username": "teleport-admin"
}
```

Resolves https://github.com/gravitational/teleport/issues/13267.

### CODE REVIEW

File: api/proto/teleport/legacy/client/proto/authservice.proto [hunk 1]
```diff
@@ -601,6 +601,11 @@ message DeleteAllApplicationServersRequest {
   string Namespace = 1 [(gogoproto.jsontag) = "namespace"];
 }
 
+// ListOfStrings is a generic list of strings.
+message ListOfStrings {
```
Comment (zmb3) [inline api/proto/teleport/legacy/client/proto/authservice.proto:605]: Use wrappers.StringValues instead of duplicating it?

Comment (mdwn) [reply api/proto/teleport/legacy/client/proto/authservice.proto:605]: Ahhh, thank you. I was looking for something like that!

File: integration/appaccess/jwt.go [hunk 2]
```diff
@@ -52,4 +53,12 @@ func verifyJWT(t *testing.T, pack *Pack, token, appURI string) {
 	require.NoError(t, err)
 	require.Equal(t, pack.username, claims.Username)
 	require.Equal(t, pack.user.GetRoles(), claims.Roles)
+
+	filteredTraits := wrappers.Traits{}
+	for trait, values := range pack.user.GetTraits() {
+		if values != nil {
```
Comment (zmb3) [inline integration/appaccess/jwt.go:59]: I'd check `len(values) > 0` instead of nil, this will handle both nil and non-nil but empty.

Comment (mdwn) [reply integration/appaccess/jwt.go:59]: Sounds good.

**Review by zmb3 — APPROVED** [review]

**Review by gabrielcorado — APPROVED** [review]

**Comment by r0mant** [comment]
@mdwn Please don't forget to update documentation to mention that traits are passed along in JWT tokens as well: https://goteleport.com/docs/application-access/jwt/introduction/.

**Comment by mdwn** [comment]
@r0mant PR for the docs updates: https://github.com/gravitational/teleport/pull/17238

**Pull request merged**